pkg/providers/vsphere/network/reconcile.go

```diff
@@ -0,0 +1,151 @@
+// © Broadcom. All Rights Reserved.
+// The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package network
+
+import (
+	"context"
+	"strings"
+
+	"github.com/vmware/govmomi/object"
+	vimtypes "github.com/vmware/govmomi/vim25/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	netopv1alpha1 "github.com/vmware-tanzu/net-operator-api/api/v1alpha1"
+	vpcv1alpha1 "github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+	ncpv1alpha1 "github.com/vmware-tanzu/vm-operator/external/ncp/api/v1alpha1"
+
+	"github.com/vmware-tanzu/vm-operator/pkg/util/resize"
+)
+
+func ReconcileNetworkInterfaces(
+	ctx context.Context,
+	results *NetworkInterfaceResults,
+	currentEthCards object.VirtualDeviceList,
+) ([]vimtypes.BaseVirtualDeviceConfigSpec, error) {
+
+	var deviceChanges []vimtypes.BaseVirtualDeviceConfigSpec
+
+	for idx, r := range results.Results {
+		matchingIdx := FindMatchingEthCard(currentEthCards, r.Device.(vimtypes.BaseVirtualEthernetCard))
+		if matchingIdx >= 0 {
+			// Exact match. Claim it by removing the device from the current ethernet cards.
+			matchDev := currentEthCards[matchingIdx].(vimtypes.BaseVirtualEthernetCard).GetVirtualEthernetCard()
+			results.Results[idx].DeviceKey = matchDev.Key
+			results.Results[idx].MacAddress = matchDev.MacAddress
+			currentEthCards = append(currentEthCards[:matchingIdx], currentEthCards[matchingIdx+1:]...)
```

Comment:
Use [slices.Delete](https://pkg.go.dev/slices#Delete) instead? (golangci-lint has been nagging me to use the slices package)